peter-piper.js

```diff
@@ -21,5 +21,8 @@ var counter = 0;
 
 // TODO: Count the number of times an uppercase P or lowercase p character
 //       occurs in the `peterpiper` string.
+for(var i  = 0; i < peterpiper.length; i++)
+    if(peterpiper[i] == "p" || peterpiper[i] == "P")
```

Comment:
BTW, the array-like access to string characters was standardized in ES5, so it is safe to use unless you are supporting Internet Explorer 8 and below.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#Character_access
http://kangax.github.io/compat-table/es5/#Property_access_on_strings (check "Show obsolete browsers" for IE support).